components/data_comps/datm/bld/build-namelist

```diff
@@ -381,9 +381,7 @@ if ( $GRID eq "CLM_USRDAT" && $CLM_USRDAT_NAME =~ /^(UNSET|)$/ ) {
 EOF
 }
 
-(-d $DIN_LOC_ROOT)  or  die <<"EOF";
-** $ProgName - CESM inputdata root is not a directory: \"$DIN_LOC_ROOT\" **
-EOF
+(-d $DIN_LOC_ROOT)  or mkdir $DIN_LOC_ROOT;
```

Comment:
@jedwards4b The problem is that some machines require us to put DIN_LOC_ROOT in a scratch directory, in which case it gets purged every once in a while. So, DIN_LOC_ROOT has not always been created before running this script.

Aside from that, there's no reason this script shouldn't create DIN_LOC_ROOT if it doesn't exist. The configuration files told it that this was where it should find / store input data, and erroring out (for normal users) in this case is more an annoyance than anything helpful.